<launch>

  <arg name="rosbot_pro" default="false" />
  <arg name="use_gazebo" default="false" />

  <!-- Gazebo -->
  <group if="$(arg use_gazebo)">
    <include file="$(find drl_navigation)/launch/training_env.launch" />
    <include file="$(find rosbot_description)/launch/rosbot_gazebo.launch"/>
    <param name="use_sim_time" value="true" />
  </group>

  <node pkg="amcl" type="amcl" name="amcl" output="screen">
    <remap from="scan" to="scan"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="odom_model_type" value="diff-corrected"/>
    <param name="base_frame_id" value="base_link"/>
    <param name="update_min_d" value="0.1"/>
    <param name="update_min_a" value="0.2"/>
    <param name="min_particles" value="500"/>
    <param name="global_frame_id" value="map"/>
    <param name="tf_broadcast" value="true" />
    <param name="initial_pose_x" value="0.0"/>
    <param name="initial_pose_y" value="0.0"/>
    <param name="initial_pose_a" value="0.0"/>
    <param name="odom_alpha1" value="0.5"/>
    <param name="odom_alpha2" value="0.5"/>
    <param name="odom_alpha3" value="0.5"/>
    <param name="odom_alpha4" value="0.5"/>
  </node>

  <arg name="map_file" default="$(find drl_navigation)/maps/training_env_no_object_map.yaml"/>       <!-- path of map file -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" respawn="true" />
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find drl_navigation)/rviz/slam_config.rviz" />

  <rosparam command="load" file="$(find drl_navigation)/config/common_params.yaml" />
  <node pkg="drl_navigation" type="pose_publisher.py" name="pose_publisher" output="screen" />

  <include file="$(find drl_navigation)/launch/laser_filter.launch" />

</launch>